common/recommender/Baseline.js

```diff
@@ -0,0 +1,108 @@
+"use strict";
+
+const URL = require("../../lib/vendor.bundle.js").urlParse;
+const {INFINITE_SCROLL_THRESHOLD} = require("../constants");
+
+/**
+ * Score function for URLs.
+ * See tests and `scoreEntry` comments for more insight into how the score is computed.
+ *
+ * @param {Array.<URLs>} history - User history used to assign higher score to popular domains.
+ */
+class Baseline {
+  constructor(history) {
+    this.domainCounts = history.reduce(this.countDomainOccurrences, new Map());
+  }
+
+  scoreEntry(entry) {
+    const urlObj = URL(entry.url);
+    const host = URL(entry.url).host;
+    const tf = Math.max(entry.visitCount, 1);
+    const idf = Math.log(this.domainCounts.size / Math.max(1, this.domainCounts.get(host)));
+    const imageCount = entry.images ? entry.images.length : 0;
+    const isBookmarked = entry.bookmarkId > 0 ? 1 : 0;
+    const hasDescription = entry.title !== entry.description ? 1 : 0;
+
+    const score = this.decay(tf * idf, // Score
+      // Features: Age in hours, number of visits to url, url query length, number of images, is a bookmark,
+      //           has description.
+      [this.normalizeTimestamp(entry.lastVisitDate), entry.visitCount, urlObj.query.length,
+       imageCount, isBookmarked, hasDescription],
+      // Features weights: Positive values decrease the score proportional to a factor of feature * weight.
+      //                   Negative values increase score proportional to a factor of feature * weight.
+      [0.4, 0.7, 0.1, -0.4, -0.2, -0.1]);
+
+    return Object.assign({}, entry, {score}, {host});
+  }
+
+  /**
+   * Scoring function for an array of URLs.
+   *
+   * @param {Array.<URLs>} entries
+   * @returns {Array.<URLs>} sorted and with the associated score value.
+   */
+  score(entries) {
+    return entries.map(entry => this.scoreEntry(entry))
+      .filter(e => e.score > 0)
```

Comment:
The only problem with filtering our 0 scores is that sometimes, we end up with only 2 items (I noticed this happening on a kind of old profile) – maybe let's not remove stuff, just sort it to the bottom (as you're already doing), until we have a better strategy to deal with not enough items